src/object.c

```diff
@@ -1618,15 +1621,35 @@ void func_80023628() {
     func_8004A67C();
 }
 
-#pragma GLOBAL_ASM("asm/nonmatchings/object/func_80023894.s")
+void func_80023894(Object* object, s32 objectId) {
+    SideCreateInfo* sidekick;
+
+    sidekick = (SideCreateInfo*)object->createInfo;
```

Comment:
nit: probably copy-paste but this shouldn't be called sidekick but instead `createInfo` or `sideCreateInfo`